extensions/cactus-plugin-htlc-coordinator/src/test/typescript/integration/plugin-htlc-coordinator/withdraw-counterparty-endpoint.test.ts

```diff
@@ -0,0 +1,295 @@
+
+  t.comment("Get HashTimeLock contract and account allowance");
+  const { callOutput } = await connector.invokeContract({
+    contractName: TestTokenJSON.contractName,
+    keychainId,
+    signingCredential: web3SigningCredential,
+    invocationType: EthContractInvocationType.Call,
+    methodName: "allowance",
+    params: [firstHighNetWorthAccount, contractAddress],
+  });
+  t.equal(callOutput, "10", "callOutput() is 10 OK");
+
+  t.comment("Create and initialize own HTLC");
+  const ownHTLCRequest: OwnHTLCRequest = {
+    htlcPackage: HtlcPackage.BesuErc20,
+    connectorId,
+    keychainId,
+    constructorArgs: [],
+    web3SigningCredential,
+    inputAmount: 10,
+    outputAmount: 1,
+    expiration,
+    hashLock,
+    tokenAddress,
+    receiver,
+    outputNetwork: "BTC",
+    outputAddress: "1AcVYm7M3kkJQH28FXAvyBFQzFRL6xPKu8",
+    gas: estimatedGas,
+  };
+
+  const response = await api.apiV1OwnCoordinator(ownHTLCRequest);
+  t.equal(response.status, 200, "response status is 200 OK");
+  t.equal(response.data.success, true, "response sucess is true");
+  t.ok(
+    response.data,
+    "pluginHTLCCoordinator.apiV1OwnCoordinator() output is truthy OK",
+  );
+  t.ok(
+    response.data.out.transactionReceipt,
+    "pluginHTLCCoordinator.apiV1OwnCoordinator() output.transactionReceipt is truthy OK",
+  );
+
+  t.comment("Get HTLC id");
+  const responseTxId = await connector.invokeContract({
+    contractName: DemoHelperJSON.contractName,
+    keychainId,
+    signingCredential: web3SigningCredential,
+    invocationType: EthContractInvocationType.Call,
+    methodName: "getTxId",
+    params: [
+      firstHighNetWorthAccount,
+      receiver,
+      10,
+      hashLock,
+      expiration,
+      tokenAddress,
+    ],
+    gas: estimatedGas,
+  });
+
+  t.comment("Get counterparty HTLC");
+  const counterpartyHTLCRequest: CounterpartyHTLCRequest = {
+    htlcPackage: HtlcPackage.BesuErc20,
+    connectorId,
+    keychainId,
+    htlcId: responseTxId.callOutput,
+    web3SigningCredential,
+    gas: estimatedGas,
+  };
+
+  const response2 = await api.apiV1CounterpartyHTLC(counterpartyHTLCRequest);
+  t.equal(response2.status, 200, "response status is 200 OK");
+  t.equal(response2.data.success, true, "response sucess is true");
+  t.equal(response2.data.callOutput, "1", "the contract status is 1 - Active");
+
+  t.comment("Get counterparty HTLC");
+  const withdrawCounterparty: WithdrawCounterpartyRequest = {
+    htlcPackage: HtlcPackage.BesuErc20,
+    connectorId,
+    keychainId,
+    web3SigningCredential,
+    htlcId: responseTxId.callOutput,
+    secret,
+    gas: estimatedGas,
+  };
+
+  const response3 = await api.apiV1WithdrawCounterparty(withdrawCounterparty);
+  t.equal(response3.status, 200, "response status is 200 OK");
+  t.equal(response3.data.success, true, "response sucess is true");
+
+  t.comment("Get balance of receiver account");
+  const responseFinalBalance = await connector.invokeContract({
```

Comment:
@jagpreetsinghsasan Please assert more specifically the status code being 2xx and also refactor this to use the API client not the connector instance directly.